pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -883,6 +897,35 @@ func endpointDiscoveryResponse(loadAssignments []*xdsapi.ClusterLoadAssignment,
 	return out
 }
 
+func endpointAcceptMtls(endpointTlsMap *map[string]bool, push *model.PushContext, ep *model.IstioEndpoint) bool {
+	b, err := json.Marshal(ep.Labels)
+	if err != nil {
+		return true
+	}
+	labelHash := fmt.Sprintf("%x", sha256.Sum256(b))[:8]
+	key := fmt.Sprintf("%v|%v|%v", ep.Attributes.Namespace, labelHash, ep.EndpointPort)
+	accept, exists := (*endpointTlsMap)[key]
+	if exists {
+		return accept
+	}
+	out := true
+	defer func() {
```

Comment:
defer is not effective enough

Reply:
sorry, could you elaborate a bit? wouldn't this ensure the tlsMap get updated in all exits of the function? i'll write unit test to ensure anyway though.

Reply:
Sorry, the function is ok, i am saying the efficiency , since the function is called more frequently.

Reply:
defer is invoked once in this function body when exit and cache miss, right?